flocker/node/agents/blockdevice.py

```diff
@@ -1832,7 +1805,7 @@ def _calculate_deletes(self, local_node_state, configured_manifestations,
             volume = _blockdevice_volume_from_datasetid(volumes, dataset_id)
             if (volume is not None and
```

Comment:
This seems like it needs fixing: we don't want to destroy things that are in local manifestations, we want to destroy things in UNATTACHED state.